### PR TITLE
[SPARK-21481][ML][FOLLOWUP][Trivial] HashingTF use util.collection.OpenHashMap instead of mutable.HashMap

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
`HashingTF` use `util.collection.OpenHashMap` instead of `mutable.HashMap`

### Why are the changes needed?
according to `util.collection.OpenHashMap` 's doc:

> This map is about 5X faster than java.util.HashMap, while using much less space overhead.

according to performance tests like ([Simple microbenchmarks comparing Scala vs Java mutable map performance ](https://gist.github.com/pchiusano/1423303)), `mutable.HashMap` maybe more inefficient than `java.util.HashMap`

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
existing testsuites

### CODE REVIEW

**Comment by zhengruifeng** [comment]
ping @huaxingao

File: mllib/src/main/scala/org/apache/spark/ml/feature/HashingTF.scala [hunk 1]
```diff
@@ -91,20 +90,13 @@ class HashingTF @Since("3.0.0") private[ml] (
...
-      terms.foreach { term =>
-        val i = indexOf(term)
-        if (localBinary) {
-          termFrequencies(i) = 1.0
-        } else {
-          termFrequencies(i) += 1.0
-        }
-      }
-      Vectors.sparse(localNumFeatures, termFrequencies.toSeq)
+      val map = new OpenHashMap[Int, Double]()
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/ml/feature/HashingTF.scala:97]: This seems fine but is it faster than Scala's Map? the comment refers to the Java HashMap.

Comment (zhengruifeng) [reply mllib/src/main/scala/org/apache/spark/ml/feature/HashingTF.scala:97]: yes, the comment in `util.collection.OpenHashMap` only refer to java hashmap, but to my knowledge, scala hashmap is slower than java hashmap.

and there is was also a [performance test](https://gist.github.com/rxin/44657c3f40d4e4294857) between java hashmap and scala hashmap: scala hashmap < java hashmap < scala openhashmap

**Review by srowen — APPROVED** [review]

**Review by huaxingao — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Comment by zhengruifeng** [comment]
Thanks @srowen and @huaxingao for help reviewing

**Pull request closed without merging**